pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -192,15 +192,30 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			Fn:           flow.TaskFn(botanist.WaitUntilControlPlaneReady),
 			Dependencies: flow.NewTaskIDs(deployControlPlane),
 		})
-		createOrUpdateEtcdEncryptionConfiguration = g.Add(flow.Task{
+		generateEncryptionConfigurationMetaData = g.Add(flow.Task{
+			Name:         "Generating etcd encryption configuration",
+			Fn:           flow.TaskFn(botanist.GenerateEncryptionConfiguration).DoIf(enableEtcdEncryption),
+			Dependencies: flow.NewTaskIDs(deployNamespace),
+		})
+		persistETCDEncryptionConfiguration = g.Add(flow.Task{
+			Name:         "Persisting etcd encryption configuration in ShootState",
+			Fn:           flow.TaskFn(botanist.PersistEncryptionConfiguration).DoIf(enableEtcdEncryption),
+			Dependencies: flow.NewTaskIDs(deployNamespace, generateEncryptionConfigurationMetaData),
+		})
+		_ = g.Add(flow.Task{
```

Comment:
Can you add a `// TODO` comment here that this task can be removed in a future version?